ykpack/src/types.rs

```diff
@@ -386,8 +386,8 @@ pub enum CallOperand {
     Unknown, // FIXME -- Find out what else. Closures jump to mind.
 }
 
-fn sym_name_str(maybe_name: &Option<String>) -> &str {
-    &maybe_name.map_or_else(|| "<unknown>", |n| n.as_str())
+fn sym_name_str(maybe_name: Option<&String>) -> &str {
```

Comment:
Can this be `Option<&str>`? Maybe not. My head hurts.

Reply:
I guess not:
```
    = note: expected type `std::option::Option<&str>`
               found type `std::option::Option<&std::string::String>`
```

Reply:
Fair enough.

Reply:
Using `.map(|s| &*s)` at the call site should coerce `&String` into `&str`.

Reply:
I've tried a few things without joy:


## `sym_name_str(sym_name.map(|s| &*s).as_ref())`

```
error[E0308]: mismatched types
   --> ykpack/src/types.rs:406:30
    |
406 |                 sym_name_str(sym_name.map(|s| &*s).as_ref())
    |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected str, found &str
    |
    = note: expected type `std::option::Option<&str>`
               found type `std::option::Option<&&str>`
```




## `sym_name_str(sym_name.map(|s| &*s))`

```
error[E0507]: cannot move out of `*sym_name` which is behind a shared reference
   --> ykpack/src/types.rs:406:30
    |
406 |                 sym_name_str(sym_name.map(|s| &*s))
    |                              ^^^^^^^^
    |                              |
    |                              move occurs because `*sym_name` has type `std::option::Option<std::string::String>`, which does not implement the `Copy` trait
    |                              help: consider borrowing the `Option`'s content: `sym_name.as_ref()`
```

## `sym_name_str(sym_name.as_ref().map(|s| &*s))`

```
error[E0308]: mismatched types
   --> ykpack/src/types.rs:406:30
    |
406 |                 sym_name_str(sym_name.as_ref().map(|s| &*s))
    |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected str, found struct `std::string::String`
    |
    = note: expected type `std::option::Option<&str>`
               found type `std::option::Option<&std::string::String>`
```

Reply:
Maybe `.as_ref().map(|s| &**s))`? Notice the double deref.

Reply:
That works! Thanks.